src/components/Link.utils.tsx

```diff
@@ -0,0 +1,36 @@
+import { Icon } from 'design-system-react';
+import type { ReactElement } from 'react';
+
+/**
+ * Programmatically determine if a link is external to the CFPB sphere of websites
+ */
+export const isExternalLink = (targetUrl: string): boolean => {
+  let parsed;
+
+  try {
+    parsed = new URL(targetUrl);
+  } catch {
+    return false; // Internal targets will fail parsing (ex. '/home')
+  }
+
+  const externalProtocols = ['http', 'tel:', 'sms:', 'ftp:'];
+  if (externalProtocols.includes(parsed.protocol)) return true;
+
+  // Any subdomain of consumerfinance.gov or the current host
+  const isInternal = new RegExp(
+    `([\\S]*\\.)?(((consumerfinance|cf)\\.gov)|(${window.location.host}))`,
+  );
+  if (!isInternal.test(parsed.host)) return true;
```

Comment:
This function is marking emails as external, and I think we want them to be all marked internal like we [said here](https://github.com/cfpb/sbl-frontend/pull/768/files#r1809480586) (since they aren't going to a website that we don't control, but rather, it's just an email address). Maybe an early `false` return for `mailto` protocols?

<img width="716" alt="Screenshot 2024-10-22 at 5 10 48 PM" src="https://github.com/user-attachments/assets/84968349-2219-4481-b735-0ca0a60f1de2">

Reply:
Corrected.
<img width="784" alt="Screenshot 2024-10-23 at 9 48 15 AM" src="https://github.com/user-attachments/assets/c49fff71-b544-483c-b086-0f61f0ef6e4d">